### PR TITLE
Update setup script

### PR DESCRIPTION
**Pull request by tomaszrybicki**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,22 +28,30 @@ else
...
 .PHONY: init all install uninstall reinstall package clean doc
 
 init:
 	mkdir -p $(BUILD_DIR)
-	cd $(BUILD_DIR) && cmake $(SOURCE_PATH) -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) -DCMAKE_INSTALL_PREFIX=$(PREFIX)
+	cd $(BUILD_DIR) && $(CMAKE) $(SOURCE_PATH) -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) -DCMAKE_INSTALL_PREFIX=$(PREFIX)
 
 all: init
-	$(MAKE) -C $(BUILD_DIR) all
+	$(MAKE) -C $(BUILD_DIR) octf
```
Comment (mariuszbarczak) [inline Makefile]: Why we don't call all?

File: source/CMakeLists.txt [hunk 2]
```diff
@@ -17,13 +17,16 @@ set_target_properties(octf PROPERTIES LINKER_LANGUAGE CXX)
...
 target_include_directories(octf
 	PUBLIC
-	${PROTOBUF_INCLUDE_DIR}
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/octf/proto>
 	$<INSTALL_INTERFACE:${OCTF_INCLUDE_DIR}/octf/proto>
 	$<INSTALL_INTERFACE:${OCTF_INCLUDE_DIR}/octf>
 )
+target_include_directories(octf
```
Comment (mariuszbarczak) [inline source/CMakeLists.txt]: I'm not able to run iotrace because of missing libprotobuf

File: Makefile (hunk 1 above)
Comment (mariuszbarczak) [reply Makefile]: And test target seemed to be not working

File: setup_dependencies.sh [hunk 3]
```diff
@@ -54,121 +73,290 @@ function detect_distribution ()
...
 
     git checkout ${latest_release}
     check_result $? "Cannot checkout version ${latest_release}"
 
+    mkdir -p ${OPT_DIR}/protobuf
+
     git submodule update --init --recursive && \
         ./autogen.sh && \
-        ./configure --prefix=/usr && \
+        ./configure --prefix=${OPT_DIR}/protobuf && \
```
[the remark this answers is not in the document]
Comment (tomaszrybicki) [reply setup_dependencies.sh]: This way protobuf would be installed like so:
/opt/octf/bin/protoc

Whereas now it is installed like so:
/opt/octf/protobuf/bin/protoc

We should discuss both of these options.

[the remark this answers is not in the document]
Comment (mariuszbarczak) [reply setup_dependencies.sh]: OK. So up to now, let's keep. Once appear a new library will re-think it.

**Pull request merged**